### ISSUE TITLE
Weird code duplication

### ISSUE THREAD

**Issue by jrvidal** [rust-lang-nursery/rustfmt#1833]
I haven't double-checked for duplicates, so I apologize in advance if this has already been reported:

```
$ rustup run nightly rustc --version
rustc 1.20.0-nightly (598eddf4f 2017-07-24)
$ rustup run nightly rustfmt --version
0.2.1-nightly ( )
```

**Input**:
```rust
trait Y where Self: Sized {
}
```
**Output**:
```rust
trait Y
trait Y
where
    Self: Sized
{
}
```

**EDIT**: updated to simpler example.
**EDIT 2**: added compiler version.

**Comment by jrvidal** [comment]
Tested on master (f0cc060b4147ad7cfc6666dc1986adb213a0ceaf) with identical results.

**Comment by topecongiro** [comment]
Thank you for filing an issue! This should be fixed via https://github.com/rust-lang-nursery/rustfmt/pull/1826.

**Comment by jrvidal** [comment]
`generics` in [`item::format_trait`](https://github.com/rust-lang-nursery/rustfmt/blob/f0cc060b4147ad7cfc6666dc1986adb213a0ceaf/src/items.rs#L906):
```
Generics { lifetimes: [], ty_params: [], where_clause: WhereClause { id: NodeId(4294967295), predicates: [BoundPredicate(WhereBoundPredicate { span: Span { lo: BytePos(14), hi: BytePos(25), ctxt: #0 }, bound_lifetimes: [], bounded_ty: type(Self), bounds: [TraitTyParamBound(PolyTraitRef { bound_lifetimes: [], trait_ref: TraitRef { path: path(Sized), ref_id: NodeId(4294967295) }, span: Span { lo: BytePos(20), hi: BytePos(25), ctxt: #0 } }, None)] })] }, span: Span { lo: BytePos(0), hi: BytePos(0), ctxt: #0 } }
```

That ` Span { lo: BytePos(0), hi: BytePos(0) }` looks like the culprit. Upstream issue?

**Comment by jrvidal** [comment]
Ooops, I didn't see your comment!

> This should be fixed via #1826

👍
